scripts/python/services/GithubService.py

```diff
@@ -160,3 +160,17 @@ def amend_team_permissions_for_repository(self, team_id: int, permission: str, r
             f"{self.organisation_name}/{repository_name}")
         self.github_client_core_api.get_organization(self.organisation_name).get_team(
             team_id).update_team_repository(repo, permission)
+
+    def get_team_id_from_team_name(self, team_name: str) -> int:
```

Comment:
String documentation tends to be prone to human error and drifts easily from what the code is actually doing (since it can't be tested and has to be updated with any change to the function).

I believe that the functions are small enough to be self-describing through the naming, type annotations on parameters and return values.

Type annotations then have the benefit of being, to some degree, testable.

Additionally, using the type annotations we could auto-generate most of the manual string docs, if we ever needed to 😄 

Plus with the presence of unit tests, they also act as a form of documentation, but documentation that now breaks when we change things! 🔥 🚒 

Happy to discuss this further though! As I'm sure much of this is my own personal bias 😂

Reply:
Omitting the documentation also then helps highlight sensitive areas, such as the `RateLimit Decorator` function where the string doc is present! 

I believe that there's more chance of people reading docs in a codebase if they are used sparingly, else people tend to start automatically filtering them out without realising 😬

Reply:
Nice justification cheers